src/parser/utils.rs

```diff
@@ -219,6 +220,55 @@ where
     Term::RecRecord(static_map, dynamic_fields, attrs)
 }
 
+/// Merge two fields by performing the merge of both their value and MetaValue if any.
+fn merge_field(rterm1: RichTerm, rterm2: RichTerm) -> RichTerm {
+    let term1 = if let Term::MetaValue(meta) = *rterm1.term {
+        (Some(meta.clone()), meta.value.clone())
+    } else {
+        (None, Some(rterm1))
+    };
+
+    let term2 = if let Term::MetaValue(meta) = *rterm2.term {
+        (Some(meta.clone()), meta.value.clone())
+    } else {
+        (None, Some(rterm2))
+    };
+
+    let new_value = if let Some(t1) = term1.1 {
+        if let Some(t2) = term2.1 {
+            mk_term::op2(BinaryOp::Merge(), t1, t2)
+        } else {
+            t1
+        }
+    } else {
+        if let Some(t2) = term2.1 {
+            t2
+        } else {
+            RichTerm::from(Term::Null)
```

Comment:
I made the function return an `Option<RichTerm>` which I unwrap in the `insert_static_field` as it's impossible to get a field with `value = None` AND `metavalue = None`.

I didn't unwrap it inside the function as it could be reused for other usages later from which this particular case may not be impossible (I guess ?)